pkg/hooks/exec/runtimeconfigfilter.go

```diff
@@ -0,0 +1,33 @@
+package exec
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"time"
+
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// RuntimeConfigFilter calls a series of hooks.  But instead of
+// passing container state on their standard input,
+// RuntimeConfigFilter passes the proposed runtime configuration (and
+// reads back a possibly-altered form from their standard output).
+func RuntimeConfigFilter(ctx context.Context, hooks []spec.Hook, config *spec.Spec, postKillTimeout time.Duration) (hookErr, err error) {
+	data, err := json.Marshal(config)
+	for _, hook := range hooks {
+		var stdout bytes.Buffer
+		hookErr, err = Run(ctx, &hook, data, &stdout, nil, postKillTimeout)
+		if err != nil {
+			return hookErr, err
+		}
+
+		data = stdout.Bytes()
```

Comment:
I feel like it would be nice to have debug-level logging of any changes here, but we don't want to bother with changes that are just whitespace (e.g. indenting or dedenting).  And we don't currently have runtime diff-generating dependencies (at least, not that I found in a quick `vendor` skim).  So I've left that off for now.  Thoughts?

Reply:
I know that our test utility can generate very nice pretty-printed diffs of structs (testify's Assert package). After a quick browse of their source tree, it seems like they are using pmezard/go-difflib to generate string diffs of structs.

Reply:
> I know that our test utility can generate very nice pretty-printed diffs of structs...

But they're currently only a test dependency, not a runtime dependency.  Are you ok with me adding go-difflib as a runtime dependency to get this logging?

Reply:
Sure - ought to be fine

Reply:
I've [added the logging][1].

[1]: https://github.com/containers/libpod/pull/1830#issuecomment-440055707